Objects/iterobject.c

```diff
@@ -316,7 +316,45 @@ anextawaitable_traverse(anextawaitableobject *obj, visitproc visit, void *arg)
 static PyObject *
 anextawaitable_iternext(anextawaitableobject *obj)
 {
-    PyObject *result = PyIter_Next(obj->wrapped);
+    assert(obj->wrapped != NULL);
+    PyTypeObject *type = Py_TYPE(obj->wrapped);
+    /* Consider the following class:
+     *
+     *     class A:
+     *         async def __anext__(self):
+     *             ...
+     *     a = A()
+     *
+     * Then anext(a) should call
+     * a.__anext__().__await__().__next__()
+     *
+     * On the other hand, given
+     *
+     *     async def agen():
+     *         yield 1
+     *         yield 2
+     *     gen = agen()
+     *
+     * Then anext(g) can just call
+     * g.__anext__().__next__()
```

Comment:
Nit: The generator above is named `gen`, not `g`.
```suggestion
     * Then anext(gen) can just call
     * gen.__anext__().__next__()
```